### PR TITLE
Match feed url when looking for feeds.

### LINKED ISSUE #944: v5: "We are still reviewing your account. Please try again later"
### Describe the bug:
Sometimes the API rejects users from connecting the feed profile.

We should shown an error message then to the user of why feed profile registration failed

### Steps to reproduce:

1. Connect a new pinterest user. You may need to disconnect the user and connect again sometimes

### Expected behavior:
When API errors, we show the error message to user so they can easily pass it back to customer support.

### Actual behavior:
App swallows the error

### Additional details:
Screenshot 2024-03-18 at 3 06 33 PM

Body: {"items":[{"preferred_processing_schedule":null,"location":"https://joyfully-chief-kingdom.jurassic.ninja/wp-content/uploads/pinterest-for-woocommerce-fpRZXX.xml","updated_at":"2024-03-18T21:58:18.017000","default_locale":"en-US","format":"XML","status":"ACTIVE","credentials":null,"name":null,"default_currency":"USD","default_country":"US","default_availability":null,"created_at":"2024-03-18T21:58:18.017000","id":"1553650766411","catalog_type":"RETAIL"}],"bookmark":null}
2024-03-18T22:03:41+00:00 Debug PATCH Request: https://api.pinterest.com/v5/catalogs/feeds/1553650766411?ad_account_id=549767509140

{"location":"https:\/\/joyfully-chief-kingdom.jurassic.ninja\/wp-content\/uploads\/pinterest-for-woocommerce-iePdXY.xml","status":"ACTIVE","preferred_processing_schedule":{"time":"22:08","timezone":"Etc\/UTC"},"catalog_type":"RETAIL"}
2024-03-18T22:03:41+00:00 Debug Response: 

Status: 403 Forbidden

Headers: {"content-type":"application\/json","content-length":"85","pinterest-version":"974758f","x-content-type-options":"nosniff","x-frame-options":"DENY","access-control-allow-credentials":"true","p3p":"CP=\"Pinterest does not have a P3P policy. You can find our privacy policy at https:\/\/www.pinterest.com\/_\/_\/policy\/privacy-policy.\"","x-envoy-upstream-service-time":"73","pinterest-generated-by":"tpp-api-prod-0a03d4e5","x-ratelimit-limit":"100, 100;w=1, 100;w=60","x-ratelimit-remaining":"98","x-ratelimit-reset":"1","cache-control":"no-cache, no-store, must-revalidate","x-pinterest-rid":"3261678879571796","date":"Mon, 18 Mar 2024 22:03:41 GMT","alt-svc":"h3=\":443\"; ma=600","set-cookie":"_ir=0; Max-Age=1800; HttpOnly; Path=\/; Secure","akamai-grn":"0.9fed19b8.1710799421.3281a9dd","x-cdn":"akamai"}

Body: {"code":2626,"message":"We are still reviewing your account. Please try again later"}
2024-03-18T22:03:41+00:00 Error We are still reviewing your account. Please try again later
#0 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/Base.php(93): Automattic\WooCommerce\Pinterest\API\Base::handle_request(Array)
#1 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/APIV5.php(531): Automattic\WooCommerce\Pinterest\API\Base::make_request('catalogs/feeds/...', 'PATCH', Array)
#2 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/Feeds.php(202): Automattic\WooCommerce\Pinterest\API\APIV5::update_feed('1553650766411', Array, '549767509140')
#3 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/FeedRegistration.php(147): Automattic\WooCommerce\Pinterest\Feeds::update_feed('1553650766411', Array)
#4 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/FeedRegistration.php(92): Automattic\WooCommerce\Pinterest\FeedRegistration::register_feed()
#5 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): Automattic\WooCommerce\Pinterest\FeedRegistration->handle_feed_registration()
#6 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#7 /srv/htdocs/__wp__/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#8 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/actions/ActionScheduler_Action.php(56): do_action_ref_array('pinterest-for-w...', Array)
#9 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php(88): ActionScheduler_Action->execute()
#10 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(169): ActionScheduler_Abstract_QueueRunner->process_action(91, 'Async Request')
#11 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(139): ActionScheduler_QueueRunner->do_batch(25, 'Async Request')
#12 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): ActionScheduler_QueueRunner->run('Async Request')
#13 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#14 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#15 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_AsyncRequest_QueueRunner.php(52): do_action('action_schedule...', 'Async Request')
#16 /srv/htdocs/wp-content/plugins/woocommerce/includes/libraries/wp-async-request.php(147): ActionScheduler_AsyncRequest_QueueRunner->handle()
#17 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): WP_Async_Request->maybe_handle('')
#18 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#19 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#20 /srv/htdocs/__wp__/wp-admin/admin-ajax.php(188): do_action('wp_ajax_as_asyn...')
#21 {main}
2024-03-18T22:03:41+00:00 Error We are still reviewing your account. Please try again later

### PR DESCRIPTION
**Pull request by budzanowski**
Match url when looking for feed.

Fixes: #944

### CODE REVIEW

**Comment by ecgan** [comment]
I tested this PR with a new Pinterest account that is not yet approved. 

At first, it seems to be working as expected. I do not see the `"We are still reviewing your account. Please try again later"` error message as per issue https://github.com/woocommerce/pinterest-for-woocommerce/issues/944 anymore (because we have removed the `Feeds::update_feed` call in this PR).

Because it is a new unapproved Pinterest account, when I visit Pinterest, I do not see the Products page:

image

However, in the plugin's Catalog page, I see the following which seems to suggest everything is fine:

image

Because of the discrepancy between the two screenshots above, I disconnected my Pinterest account, deactivate and delete the plugin, reinstall the plugin again, and reconnect the same Pinterest account.

This time, I see the following "We are still reviewing your account" error message in log file: (notice that it is a "delete feed" request, not "update feed" request)

```log
2024-03-19T15:48:59+00:00 Debug DELETE Request: https://api.pinterest.com/v5/catalogs/feeds/1547252346792?ad_account_id=549767511953

--- EMPTY STRING ---
2024-03-19T15:49:00+00:00 Debug Response: 

Status: 403 Forbidden

Headers: {"content-length":"85","content-type":"application\/json","pinterest-version":"974758f","x-content-type-options":"nosniff","x-frame-options":"DENY","access-control-allow-credentials":"true","p3p":"CP=\"Pinterest does not have a P3P policy. You can find our privacy policy at https:\/\/www.pinterest.com\/_\/_\/policy\/privacy-policy.\"","x-envoy-upstream-service-time":"68","x-ratelimit-limit":"100, 100;w=1, 100;w=60","x-ratelimit-remaining":"99","x-ratelimit-reset":"1","cache-control":"no-cache, no-store, must-revalidate","age":"0","set-cookie":"_ir=0; Max-Age=1800; HttpOnly; Path=\/; Secure","x-pinterest-rid":"1211442572000082","date":"Tue, 19 Mar 2024 15:49:00 GMT","x-cdn":"fastly","alt-svc":"h3=\":443\";ma=600","pinterest-generated-by":"tpp-api-canary-0a0308c3"}

Body: {"code":2626,"message":"We are still reviewing your account. Please try again later"}
2024-03-19T15:49:00+00:00 Error We are still reviewing your account. Please try again later
#0 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/Base.php(93): Automattic\WooCommerce\Pinterest\API\Base::handle_request(Array)
#1 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/APIV5.php(509): Automattic\WooCommerce\Pinterest\API\Base::make_request('catalogs/feeds/...', 'DELETE')
#2 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/Feeds.php(340): Automattic\WooCommerce\Pinterest\API\APIV5::delete_feed('1547252346792', '549767511953')
#3 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/FeedRegistration.php(179): Automattic\WooCommerce\Pinterest\Feeds::delete_feed('1547252346792')
#4 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/class-pinterest-for-woocommerce.php(802): Automattic\WooCommerce\Pinterest\FeedRegistration::maybe_delete_stale_feeds_for_merchant('')
#5 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/AuthDisconnect.php(45): Pinterest_For_Woocommerce::disconnect()
#6 /wordpress/core/6.4.3/wp-includes/rest-api/class-wp-rest-server.php(1193): Automattic\WooCommerce\Pinterest\API\AuthDisconnect->handle_disconnect(Object(WP_REST_Request))
#7 /wordpress/core/6.4.3/wp-includes/rest-api/class-wp-rest-server.php(1041): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '/pinterest/v1/a...', Array, NULL)
#8 /wordpress/core/6.4.3/wp-includes/rest-api/class-wp-rest-server.php(431): WP_REST_Server->dispatch(Object(WP_REST_Request))
#9 /wordpress/core/6.4.3/wp-includes/rest-api.php(424): WP_REST_Server->serve_request('/pinterest/v1/a...')
#10 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): rest_api_loaded(Object(WP))
#11 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#12 /wordpress/core/6.4.3/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#13 /wordpress/core/6.4.3/wp-includes/class-wp.php(418): do_action_ref_array('parse_request', Array)
#14 /wordpress/core/6.4.3/wp-includes/class-wp.php(813): WP->parse_request('')
#15 /wordpress/core/6.4.3/wp-includes/functions.php(1336): WP->main('')
#16 /wordpress/core/6.4.3/wp-blog-header.php(16): wp()
#17 /wordpress/core/6.4.3/index.php(17): require('/wordpress/core...')
#18 {main}
2024-03-19T15:49:00+00:00 Error We are still reviewing your account. Please try again later
```

In the Catalog page, I see that things are pending, which match with what I see in Pinterest website (no Products page):

image

There is also another error message "The feed name already exists." in my log file:

```
2024-03-19T15:51:06+00:00 Debug POST Request: https://api.pinterest.com/v5/catalogs/feeds?ad_account_id=549767511953

{"name":"Created by Pinterest for WooCommerce US|en-US|USD","format":"XML","location":"https:\/\/ganpfwv5.mystagingwebsite.com\/wp-content\/uploads\/pinterest-for-woocommerce-SWzPuV.xml","catalog_type":"RETAIL","default_currency":"USD","default_locale":"en-US","default_country":"US","default_availability":"IN_STOCK","preferred_processing_schedule":{"time":"15:56","timezone":"Etc\/UTC"}}
2024-03-19T15:51:06+00:00 Debug Response: 

Status: 422 Unprocessable Entity

Headers: {"content-length":"55","content-type":"application\/json","pinterest-version":"974758f","x-content-type-options":"nosniff","x-frame-options":"DENY","access-control-allow-credentials":"true","p3p":"CP=\"Pinterest does not have a P3P policy. You can find our privacy policy at https:\/\/www.pinterest.com\/_\/_\/policy\/privacy-policy.\"","x-envoy-upstream-service-time":"72","x-ratelimit-limit":"100, 100;w=1, 100;w=60","x-ratelimit-remaining":"98","x-ratelimit-reset":"1","cache-control":"no-cache, no-store, must-revalidate","age":"0","set-cookie":"_ir=0; Max-Age=1800; HttpOnly; Path=\/; Secure","x-pinterest-rid":"1521448715364458","date":"Tue, 19 Mar 2024 15:51:06 GMT","x-cdn":"fastly","alt-svc":"h3=\":443\";ma=600","pinterest-generated-by":"tpp-api-prod-0a038a91"}

Body: {"code":4170,"message":"The feed name already exists."}
2024-03-19T15:51:06+00:00 Error The feed name already exists.
#0 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/Base.php(93): Automattic\WooCommerce\Pinterest\API\Base::handle_request(Array)
#1 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/API/APIV5.php(492): Automattic\WooCommerce\Pinterest\API\Base::make_request('catalogs/feeds?...', 'POST', Array)
#2 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/Feeds.php(150): Automattic\WooCommerce\Pinterest\API\APIV5::create_feed(Array, '549767511953')
#3 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/FeedRegistration.php(139): Automattic\WooCommerce\Pinterest\Feeds::create_feed()
#4 /srv/htdocs/wp-content/plugins/pinterest-for-woocommerce/src/FeedRegistration.php(92): Automattic\WooCommerce\Pinterest\FeedRegistration::register_feed()
#5 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): Automattic\WooCommerce\Pinterest\FeedRegistration->handle_feed_registration()
#6 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#7 /wordpress/core/6.4.3/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#8 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/actions/ActionScheduler_Action.php(56): do_action_ref_array('pinterest-for-w...', Array)
#9 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php(88): ActionScheduler_Action->execute()
#10 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(169): ActionScheduler_Abstract_QueueRunner->process_action(294, 'WP Cron')
#11 /srv/htdocs/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(139): ActionScheduler_QueueRunner->do_batch(25, 'WP Cron')
#12 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): ActionScheduler_QueueRunner->run('WP Cron')
#13 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#14 /wordpress/core/6.4.3/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#15 /wordpress/core/6.4.3/wp-cron.php(191): do_action_ref_array('action_schedule...', Array)
#16 {main}
```

**Comment by ecgan** [comment]
Given that this PR fixes the "update feed" issue, I think we can proceed to review, approve and merge this PR. However, note that this PR does not really fix https://github.com/woocommerce/pinterest-for-woocommerce/issues/944 though, since there are other errors in the log file. We can open separate PRs to fix the "delete feed" and "The feed name already exists." issues.
